src/component/axis/AxisBuilder.ts

```diff
@@ -372,7 +372,7 @@ const builders: Record<'axisLine' | 'axisTickLabel' | 'axisName', AxisElementsBu
         const textStyleModel = axisModel.getModel('nameTextStyle');
         const gap = axisModel.get('nameGap') || 0;
 
-        const extent = axisModel.axis.getExtent();
+        const extent = axisModel.axis.getGridExtent();
         const gapSignal = extent[0] > extent[1] ? -1 : 1;
```

Comment:
Sorry for this late reply
The reason why `splitline` of two axis cannot align in most case, it's because axis will fill the `grid`[code](https://github.com/apache/incubator-echarts/blob/next/src/coord/cartesian/Grid.ts#L166).  `axis` will be split evenly by using `grid extent`. What I trying to do is calculate the `proportion` of axis in this grid, each axis has same standard which is `unionExtent`.

Reply:
@pissang So, if user specified `min` or `max`, `axis extent` will be different from `grid extent`.

Reply:
I thought the main reason two axes don't align is that they don't have the same split numbers. They should be aligned if  the split numbers are same.

Reply:
@pissang That's correct, because without `min` `max`, axis will fill the `grid`, the proportion will be `100%`, with `min` or `max`, it will be less than `100%`(in most case, some case can still have `100%`, `max % interval === 0` ).